build-tools/packages/version-tools/src/test/internalVersionScheme.test.ts

```diff
@@ -387,4 +388,60 @@ describe("internalScheme", () => {
 			);
 		});
 	});
+
+	describe("checking test version scheme", () => {
+		it("0.0.0-285010-test is a valid test version", () => {
+			const input = `0.0.0-285010-test`;
+			const result = isInternalTestVersion(input);
+			assert.isTrue(result);
+		});
+
+		it("0.0.0-test-285010 is not a valid test version", () => {
+			const input = `0.0.0-test-285010`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0-test-285010 is not a valid test version", () => {
+			const input = `2.1.0-test-285010`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0-285010-test is not test version", () => {
+			const input = `2.1.0-285010-test`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0-285010 is a prerelease version", () => {
+			const input = `2.1.0-285010`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0 is a release version", () => {
```

Comment:
Some of these test descriptions seems wrong.

Reply:
Is there anything specific that stands out to you? probably 2.1.0 is a public version?

Reply:
So, the name/description structure of the tests in here could be improved IMO. I can give a lot of them a pass, but it leads to questionable additional descriptions like this which appear to be following the pattern.
The best test naming patterns tell you what is being tested (usually an action) under what conditions and the expectation. The full name for this case is "internalScheme > checking test version scheme > 2.1.0 is a release version". That doesn't actually indicate whether the scheme is detected as test version or not.
My suggestion for the case naming before could have been more explicit. For this one, you could say "release versions (like 2.1.0) are not test versions" or you could say '2.1.0 (a "release" version) is not a test version'.